pkg/kubectl/cmd/util/helpers.go

```diff
@@ -428,3 +429,39 @@ func UpdateObject(info *resource.Info, updateFn func(runtime.Object) error) (run
 
 	return info.Object, nil
 }
+
+// isClusterService checks the resource's label to see if it's a cluster service (add-on)
+func isClusterService(info *resource.Info) (bool, error) {
+	if info.Object == nil {
+		info.Get()
+	}
+	labels, err := info.Mapping.MetadataAccessor.Labels(info.Object)
+	if err != nil {
+		return false, err
+	}
+	if value, ok := labels[AddOnLabel]; ok && value == "true" {
+		return true, nil
+	}
+	return false, nil
+}
+
+// MutateClusterServiceError returns an error when the input resource is a cluster service (add-on)
+// This function should only be called before mutating an existing resource
+func MutateClusterServiceError(info *resource.Info) error {
+	if isAddon, err := isClusterService(info); err != nil {
+		return err
+	} else if isAddon {
+		return fmt.Errorf("%q is a cluster service and should not be mutated. See http://releases.k8s.io/HEAD/cluster/addons for more info.", info.Name)
+	}
+	return nil
+}
+
+// AddMutateClusterServiceFlag adds a flag that prevents the manipulation on add-ons. Used by mutations only.
+func AddMutateClusterServiceFlag(cmd *cobra.Command) {
+	cmd.Flags().BoolP("addon-check", "", true, "If true, prevents the users from manipulating add-ons / cluster services. See http://releases.k8s.io/HEAD/cluster/addons for more info. Default true.")
```

Comment:
deletion protection seems like something we'd want in general, and this seems like a too-specific application of it... can we make this flag more general?